src/main/java/org/jenkinsci/extension_indexer/SourceAndLibs.java

```diff
@@ -185,7 +185,9 @@ private static void downloadDependencies(File pomDir, File destDir) throws IOExc
         command.addAll(Arrays.asList("--settings", (System.getenv("MAVEN_SETTINGS") != null) ? System.getenv("MAVEN_SETTINGS") : new File("maven-settings.xml").getAbsolutePath()));
         command.addAll(Arrays.asList("--update-snapshots",
                 "--batch-mode",
-                "dependency:copy-dependencies",
+                "org.apache.maven.plugins:maven-dependency-plugin:3.8.0:copy-dependencies",
+                "org.apache.maven.plugins:maven-dependency-plugin:3.8.0:copy",
+                "-Dartifact=org.jenkins-ci.main:jenkins-core:2.479",
```

Comment:
Tested in https://github.com/timja-org/backend-extension-indexer/pull/3

could revert back to 2.480 although this will demonstrate it working.

Reply:
Noting https://github.com/jenkins-infra/backend-extension-indexer/pull/95#issuecomment-2408564440 for visibility just in case.

Reply:
Yes would be better, bit of spaghetti though, looks like it always uses the latest weekly version for other areas.

I got fairly far along on an approach:

```diff
diff --git a/src/main/java/org/jenkinsci/extension_indexer/ExtensionPointListGenerator.java b/src/main/java/org/jenkinsci/extension_indexer/ExtensionPointListGenerator.java
index 5115491..d01fa16 100644
--- a/src/main/java/org/jenkinsci/extension_indexer/ExtensionPointListGenerator.java
+++ b/src/main/java/org/jenkinsci/extension_indexer/ExtensionPointListGenerator.java
@@ -210,7 +210,8 @@ public class ExtensionPointListGenerator {
         if (asciidocOutputDir ==null && jsonFile==null && pluginsDir ==null)
             throw new IllegalStateException("Nothing to do. Either -adoc, -json, or -pipeline is needed");
 
-        discover(addModule(new Module.CoreModule(updateCenterJson.getJSONObject("core").getString("version"))));
+        Module.CoreModule coreModule = new Module.CoreModule(updateCenterJson.getJSONObject("core").getString("version"));
+        discover(addModule(coreModule), coreModule);
 
         processPlugins(updateCenterJson.getJSONObject("plugins").values());
 
@@ -331,9 +332,9 @@ public class ExtensionPointListGenerator {
         }
     }
 
-    private void discover(Module m) throws IOException, InterruptedException {
+    private void discover(Module m, Module.CoreModule core) throws IOException, InterruptedException {
         if (asciidocOutputDir !=null || jsonFile!=null) {
-            for (ClassOfInterest e : extractor.extract(m)) {
+            for (ClassOfInterest e : extractor.extract(m, core)) {
                 synchronized (families) {
                     System.out.println("Found "+e);
 
diff --git a/src/main/java/org/jenkinsci/extension_indexer/ExtensionPointsExtractor.java b/src/main/java/org/jenkinsci/extension_indexer/ExtensionPointsExtractor.java
index 131518e..8242a86 100644
--- a/src/main/java/org/jenkinsci/extension_indexer/ExtensionPointsExtractor.java
+++ b/src/main/java/org/jenkinsci/extension_indexer/ExtensionPointsExtractor.java
@@ -36,8 +36,8 @@ import java.util.Map;
  * @author Kohsuke Kawaguchi
  */
 public class ExtensionPointsExtractor {
-    public List<ClassOfInterest> extract(Module module) throws IOException, InterruptedException {
-        return extract(module,SourceAndLibs.create(module));
+    public List<ClassOfInterest> extract(Module module, Module.CoreModule core) throws IOException, InterruptedException {
+        return extract(module,SourceAndLibs.create(module, core));
     }
 
     public List<ClassOfInterest> extract(final Module module, final SourceAndLibs sal) throws IOException {
diff --git a/src/main/java/org/jenkinsci/extension_indexer/SourceAndLibs.java b/src/main/java/org/jenkinsci/extension_indexer/SourceAndLibs.java
index 58ed4d9..c9870e4 100644
--- a/src/main/java/org/jenkinsci/extension_indexer/SourceAndLibs.java
+++ b/src/main/java/org/jenkinsci/extension_indexer/SourceAndLibs.java
@@ -144,7 +144,7 @@ public class SourceAndLibs implements Closeable {
         return conn;
     }
 
-    public static SourceAndLibs create(Module module) throws IOException, InterruptedException {
+    public static SourceAndLibs create(Module module, Module.CoreModule core) throws IOException, InterruptedException {
         final File tempDir = Files.createTempDirectory("jenkins-extPoint").toFile();
         File srcdir = new File(tempDir,"src");
         File libdir = new File(tempDir,"lib");
@@ -163,7 +163,7 @@ public class SourceAndLibs implements Closeable {
         }
 
         System.out.println("Downloading Dependencies");
-        downloadDependencies(srcdir, libdir);
+        downloadDependencies(srcdir, libdir, core);
 
         return new SourceAndLibs(srcdir, libdir) {
             @Override
@@ -174,7 +174,7 @@ public class SourceAndLibs implements Closeable {
     }
 
     @SuppressFBWarnings(value = "COMMAND_INJECTION", justification = "Command injection is not a viable risk here")
-    private static void downloadDependencies(File pomDir, File destDir) throws IOException, InterruptedException {
+    private static void downloadDependencies(File pomDir, File destDir, Module.CoreModule core) throws IOException, InterruptedException {
         Files.createDirectories(destDir.toPath());
         String process = "mvn";
         if (System.getenv("M2_HOME") != null) {
@@ -183,11 +183,12 @@ public class SourceAndLibs implements Closeable {
         List<String> command = new ArrayList<>();
         command.add(process);
         command.addAll(Arrays.asList("--settings", (System.getenv("MAVEN_SETTINGS") != null) ? System.getenv("MAVEN_SETTINGS") : new File("maven-settings.xml").getAbsolutePath()));
+
         command.addAll(Arrays.asList("--update-snapshots",
                 "--batch-mode",
                 "org.apache.maven.plugins:maven-dependency-plugin:3.8.0:copy-dependencies",
                 "org.apache.maven.plugins:maven-dependency-plugin:3.8.0:copy",
-                "-Dartifact=org.jenkins-ci.main:jenkins-core:2.479",
+                "-Dartifact=" + core.gav,
                 "-DincludeScope=compile",
                 "-DoutputDirectory=" + destDir.getAbsolutePath()));
 

```